include/odp/api/spec/packet_io_stats.h

```diff
@@ -178,6 +288,44 @@ typedef struct odp_pktio_stats_capability_t {
  */
 int odp_pktio_stats(odp_pktio_t pktio, odp_pktio_stats_t *stats);
 
+/**
+ * Get statistics for packet input queue
+ *
+ * Input queue handle type must match the configured packet input mode
+ * (odp_pktin_mode_t). Valid handles can be fetched with odp_pktin_queue() and
+ * odp_pktin_event_queue() calls.
+ *
+ * Counters not supported by the interface are set to zero.
+ *
+ * @param       pktio	 Packet IO handle
+ * @param       queue	 Input queue handle
+ * @param[out]  stats	 Output buffer for counters
+ *
+ * @retval  0 on success
+ * @retval <0 on failure
+ */
+int odp_pktin_queue_stats(odp_pktio_t pktio, odp_pktio_queue_t queue,
+			  odp_pktin_queue_stats_t *stats);
```

Comment:
In this case, it would be more logical to pass "odp_pktin_queue_t queue". Unless, the entire PKTIO API is realigned with using unified queue descriptor, so the usage of descriptors is uniform. The same applies to the pktout queue statistics call.

Reply:
We'd need a separate function for packet input mode `ODP_PKTIN_MODE_SCHED` and `ODP_PKTIN_MODE_QUEUE`.

Reply:
Fixed in V4.